mautic/mautic
# Reports: Group by date

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | Pending
| Related developer documentation PR URL | Pending
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR both adds a feature and fixes a bug.

##### Feature
Previously, when grouping by a datetime field, the results from the report were... unimpressive to say the least. Since the grouping would always take into account the value of the field down to the microsecond (in PHP 7.1+, to the second in previous versions), the data didn't offer much insight into the number of emails sent out per day. There was a widget that made that specific data available, but now you can build it into your report and access that metric outside of just email sends per day.

[//]: # ( As applicable: )
#### Steps to test this PR:
1. Create an email report with `Date Sent` as the column, `Date Sent` as the `Group by`, and add a calculated field of `COUNT` on `Contact email`. 
2. On the `Graphs` tab, add the `Emails sent (line graph)`
3. Save and Close
4. This should produce a report that gives you the total number of contacts reached per day
5. Verify that the report counts are correct by validating the values in the `COUNT Contact Email` column with the values in the `Emails sent` graph.
